front/app/utils/paginationUtils.ts

```diff
@@ -1,13 +1,18 @@
-import { parse } from 'qs';
-import { isNil } from 'lodash-es';
+import { parse } from 'query-string';
+import Url from 'url-parse';
 
 export type SortDirection = 'ascending' | 'descending';
 
-export function getPageNumberFromUrl(url) {
-  if (!url || typeof url !== 'string' || url === '') return null;
-  const queryString = url.split('?')[1];
-  const result = parse(queryString).page?.['number'];
-  return isNil(result) || result < 0 ? null : parseInt(result, 10);
+export function getPageNumberFromUrl(url: string) {
+  if (url === '') return null;
+
+  const queryString = new Url(url).query;
```

Comment:
We should use React Router to get the query via the router props `location -> query`